### PR TITLE
Feat 22: Binary Crossentropy Loss Implementation

### PR DESCRIPTION
**Pull request by haruiz**

### CODE REVIEW

File: elegy/losses/binary_crossentropy.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import jax.numpy as jnp
+from elegy import utils
+from elegy.losses.loss import Loss, Reduction
+
+
+def binary_cross_entropy(
+    y_true: jnp.ndarray, y_pred: jnp.ndarray, from_logits: bool = False
+) -> jnp.ndarray:
+    if from_logits:
+        y_pred = jax.nn.log_sigmoid(y_pred)
```
Comment (cgarciae) [inline elegy/losses/binary_crossentropy.py:13]: @sebasarango1180 We tried this right? I think its not the correct formula.

**Comment by cgarciae** [comment]
@sebasarango1180 can you add your tests to this branch?

Now that we launched I think we could start adding stricter tests, specifically, testing numerically against the tf.keras counterparts on e.g. random matrices.

File: elegy/losses/binary_crossentropy.py (hunk 1 above)
Comment (sebasarango1180) [reply elegy/losses/binary_crossentropy.py:13]: @cgarciae I'm validating this. And sure, I will check the testing.

File: elegy/losses/binary_crossentropy.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    y_true: jnp.ndarray, 
+    y_pred: jnp.ndarray, 
+    from_logits: bool = False
+    ) -> jnp.ndarray:
+
+    y_pred = jnp.maximum(y_pred, utils.EPSILON)
+
+    if from_logits:
+        y_pred = jax.nn.sigmoid(y_pred)
+    return -jnp.mean(y_true * jnp.log(y_pred) + (1 - y_true) * jnp.log(1 - y_pred), axis=-1)
```
Comment (cgarciae) [inline elegy/losses/binary_crossentropy.py:19]: clip `log` using `EPSILON` as donde in `CategoricalCrossentropy`

File: elegy/losses/binary_crossentropy.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+def binary_crossentropy(
+    y_true: jnp.ndarray, 
+    y_pred: jnp.ndarray, 
+    from_logits: bool = False
+    ) -> jnp.ndarray:
+
+    if from_logits:
+        return -jnp.mean(y_true * y_pred - jnp.logaddexp(0.0, y_pred), axis=-1)
+
+    y_pred = jnp.maximum(y_pred, utils.EPSILON)
```
Comment (cgarciae) [inline elegy/losses/binary_crossentropy.py:18]: use

```python
y_pred = jnp.clip(y_pred, utils.EPSILON, 1.0 - utils.EPSILON)
```

Comment (sebasarango1180) [reply elegy/losses/binary_crossentropy.py]: Does [dd998a6](https://github.com/poets-ai/elegy/pull/51/commits/dd998a6d3bca296d4be756f51e7f668c92a6a397) make it? Or is there something else missing?
Test passing.

File: elegy/losses/binary_crossentropy.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+class BinaryCrossentropy(Loss):
+    """Computes the cross-entropy loss between true labels and predicted labels.
+      Use this cross-entropy loss when there are only two label classes (assumed to
+      be 0 and 1). For each example, there should be a single floating-point value
+      per prediction.
+      In the snippet below, each of the four examples has only a single
+      floating-pointing value, and both `y_pred` and `y_true` have the shape
+      `[batch_size]`.
+      Standalone usage:
+      >>> y_true = [[0., 1.], [0., 0.]]
```
Comment (cgarciae) [inline elegy/losses/binary_crossentropy.py:32]: Remove `>>>` and add a "```python" block

**Comment by cgarciae** [comment]
We are still missing the page in the docs.

**Comment by sebasarango1180** [comment]
@cgarciae 
- Fixed docstring as requested ([48450fa](https://github.com/poets-ai/elegy/pull/51/commits/48450faf549d9bfc3fc8c63da14395a46779092c))
- Docs were updated in ([069f7b0](https://github.com/poets-ai/elegy/pull/51/commits/069f7b040094d502228708e74b709569f99203d4)), and is currently visible in the branch. Please let me know if the page is somehow missing, or if there are missing steps.

**Comment by cgarciae** [comment]
LGTM!
